File: webservices/src/main/java/fieg/modulos/cobrancaagrupada/dto/CobrancasGrupoDTO.java
package fieg.modulos.cobrancaagrupada.dto;

import fieg.modulos.cobrancacliente.enums.SituacaoCobrancaCliente;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter
@Setter
@ToString(callSuper = true)
@AllArgsConstructor
@NoArgsConstructor
public class CobrancasGrupoDTO {

    private Integer contrato;
    private LocalDateTime dataVencimento;
    private LocalDateTime dataPagamento;
    private Integer parcela;
    private SituacaoCobrancaCliente situacao;
    private String nossoNumero;
    private String cpfCnpj;
    private String sacadoNome;
    private BigDecimal valorCobranca;
    private BigDecimal totalDebito;
    private BigDecimal totalDesconto;
    private BigDecimal valorTotalParcela;
    private String notaFiscal;
    private LocalDateTime dataEmissaoNotaFiscal;
    private String avisoLancamentoNota;
    private LocalDateTime dataAvisoLancamentoNota;
    private String status;
    private Integer idSistema;
    private Integer idUnidadeContrato;
    private Integer idCobrancaAgrupada;
    private Integer idCobrancaCliente;
    private Integer idInterface;
    private LocalDateTime terminoVigenciaCobranca;
    private Integer idGrupoCancelado;


}

File: web/src/models/DTOs/CobrancasAgrupadas/CobrancasGrupoDTO.ts
import { SituacaoCobrancaCliente } from "../Cobrancas/SituacaoCobrancaCliente";
import { StatusInterface } from "../Cobrancas/StatusInterface";

export interface CobrancasGrupoDTO {
  contrato?: number;
  dataVencimento: Date;
  dataPagamento: Date;
  parcela?: number;
  situacao: SituacaoCobrancaCliente;
  nossoNumero: string;
  cpfCnpj: string;
  sacadoNome: string;
  valorCobranca: number;
  totalDebito: number;
  totalDesconto: number;
  valorTotalParcela: number;
  notaFiscal: string;
  dataEmissaoNotaFiscal: Date;
  avisoLancamentoNota: string;
  dataAvisoLancamentoNota: Date;
  status: StatusInterface;
  idSistema: number;
  idUnidadeContrato: number;
  idCobrancaAgrupada: number;
  idCobrancaCliente: number;
  idInterface: number;
  terminoVigenciaCobranca: Date;
  idGrupoCancelado: number;
}
